datafusion/expr/src/built_in_function.rs

```diff
@@ -745,9 +748,10 @@ impl BuiltinScalarFunction {
                     return plan_err!("The to_hex function can only accept integers.");
                 }
             }),
-            BuiltinScalarFunction::ToTimestamp => Ok(Timestamp(Nanosecond, None)),
+            BuiltinScalarFunction::ToTimestamp => Ok(Timestamp(Second, None)),
```

Comment:
while writing up https://github.com/apache/arrow-datafusion/issues/7868 it occured to me this signature prevents implementing something like `to_timestamp(1.234)` reasonably, as it can't represent sub-second precison

What would you think about leaving the return type the same (`Timestamp(Nanosecond, None)`) so that we can add support for floating point argments as a follow on?

Reply:
Return `Timestamp(Nanosecond, None)` seems reasonable to me since most of the time related functions in datafusion return nanosecond based timestamp i.e. `timestamp '2000-01-01T00:00:00'`, `now()`, ...

Reply:
In PG 
```
select pg_typeof(to_timestamp(1.234))
```
the return type is `timestamp with time zone` which has minimum resolution as a microsecond

https://www.postgresql.org/docs/current/datatype-datetime.html

In light of this, having the return type as nanos or micros makes more sense for me.
What do you think is better? microseconds like in PG or nanos like it was in DF before the PR?

Reply:
I don't have any preference now that we have coercion rules that can automatically coerce between timestamp resolution

Reply:
got it, I'll do micros as a return type as it is in PG, will update the PR soon

Reply:
set return type to nanos as it was before.